src/backend/distributed/metadata/distobject.c

```diff
@@ -373,3 +373,57 @@ GetDistributedObjectAddressList(void)
 
 	return objectAddressList;
 }
+
+
+/*
+ * UpdateDistributedObjectColocationId gets an old and a new colocationId
+ * and updates the colocationId of all tuples in citus.pg_dist_object which
+ * have the old colocationId to the new colocationId.
+ */
+void
+UpdateDistributedObjectColocationId(uint32 oldColocationId,
+									uint32 newColocationId)
+{
+	const bool indexOK = false;
+	ScanKeyData scanKey[1];
+	Relation pgDistObjectRel = table_open(DistObjectRelationId(),
+										  RowExclusiveLock);
+	TupleDesc tupleDescriptor = RelationGetDescr(pgDistObjectRel);
+
+	/* scan pg_dist_object for colocationId equal to old colocationId */
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_object_colocationid,
+				BTEqualStrategyNumber,
+				F_OIDEQ,
+				ObjectIdGetDatum(oldColocationId));
+
+	SysScanDesc scanDescriptor = systable_beginscan(pgDistObjectRel,
+													DistObjectPrimaryKeyIndexId(),
```

Comment:
how can we use the primary key index here? it does not contain the colocation ID

Reply:
You are right, I forgot to change that to `InvalidOid`.

(already because `indexOK` is false, `systable_beginscan` doesn't look at the `indexId` at all)